apps/verticalhome/js/pin_apps.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+(function (exports) {
+
+  function PinAppItem(index) {
+    this.index = index;
+    this.element = document.querySelector("div[data-index='" + this.index + "']");
```

Comment:
I would suggest check of incoming parameter at least for 'undefined' & 'null' values, since querySelector most probably won't return something useful for string = "div[data-index=null]"